threads/test_thread_rand.c

```diff
@@ -0,0 +1,88 @@
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	uint16_t time;
+	uint16_t id;
+} threads_self_t;
+
+
+static int cthread = 10, fin = 0;
+
+
+static char stack[10 * 512];
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+	__sync_fetch_and_add(&fin, 1);
+	endthread();
+}
+
+
+TEST_GROUP(threads_rand);
+
+
+TEST_SETUP(threads_rand)
+{
+}
+
+
+TEST_TEAR_DOWN(threads_rand)
+{
+}
+
+
+TEST(threads_rand, test_1)
+{
+	threads_self_t params[cthread];
```

Comment:
This is the inconsequence. Thread stacks are defined globally in BSS, but thread parameters are define locally on stack? Why?

Reply:
Stack was moved to BSS because it was necessary to even store it. Parameters were left as local. Now with common struct `params` they are also stored in BSS for consistency. 

Only variables `joined` and `i` were left on stack as they scope is limited only for use in main test function `TEST(threads_rand, test_1)`